hack/update-clear-linux-base.sh

```diff
@@ -45,6 +63,60 @@ docker run "$base" swupd update --version=$version || die "failed to update"
 sed -i -e 's;^\(ARG CLEAR_LINUX_BASE=\).*;\1'"$base"';' -e 's;^\(ARG SWUPD_UPDATE_ARG=\).*;\1"--version='"$version"'";' Dockerfile || die "failed to patch Dockerfile"
 
 # Show the modification
+if git diff --exit-code Dockerfile; then
+    echo "No new version, exiting now."
+    exit 0
+fi
+
+no_image_changes () (
+    image="$1"
+    tmpdir=$(mktemp -d)
+    container1=
+    container2=
+    trap 'rm -rf $tmpdir; for id in $container1 $container2; do docker rm $id; done' EXIT
+
+    # We exclude certain irrelevant files by not even unpacking them.
+    # We could also delete some of them from the layer in the first place.
+    cat >"$tmpdir/exclude" <<EOF
+usr/share/locale/**
+usr/share/debuginfo/**
+usr/share/info/**
+usr/share/man/**
+usr/share/doc/**
+var/cache/man/**
+usr/share/package-licenses/**
+*.pyc
+etc/os-release
+lib/os-release
+usr/lib/os-release
+EOF
+
+    # We only compare file content. uid/gid, timestamp and permission changes
+    # could theoretically be the only changes, but that seems unlikely.
+    # Should that ever be the only change, we'll still update eventually
+    # once a later update changes a file.
+    container1=$(docker create $image)
+    container2=$(docker create $image-updated)
+    mkdir "$tmpdir/$oldversion"
+    mkdir "$tmpdir/$version"
+    docker export $container1 | tar -C "$tmpdir/$version" -xf - --exclude-from="$tmpdir/exclude"
+    docker export $container2 | tar -C "$tmpdir/$oldversion" -xf -  --exclude-from="$tmpdir/exclude"
+    diff --brief -r --exclude-from=- "$tmpdir/$oldversion" "$tmpdir/$version" <<EOF
+$tmpdir/$oldversion/usr/share/locale/**
+EOF
+)
+
+if [ "$oldversion" != "latest" ]; then
+    # Build images again with updated Dockerfile.
+    if ( docker build --target build -t pmem-csi-build-updated . || die "failed to build the 'build' image" ) &&
```

Comment:
Where we cleaning these intermediate({runtime,build}-update) images in case of "no update needed"?

Reply:
We don't, because it is not necessary: this will run in the CI environment, which always starts from scratch with an empty Docker environment.